src/app/mapView/MapView.jsx

```diff
@@ -32,41 +25,109 @@ export default class MapView extends Component {
     navigator.geolocation.getCurrentPosition(
       userPosition => {
         const { coords } = userPosition;
-        this.onCenterChanged({
-          lat: coords.latitude,
-          lng: coords.longitude,
+        this.setState({
+          center : {
+            lat: coords.latitude,
+            lng: coords.longitude,
+          }
         });
       },
       e => console.error('Failed to get current position', e),
       { timeout: geolocationTimeout },
     );
   }
+
   onSearchChanged = event => {
-    this.setState({ searchString: event.target.value }, () => {
-      this.fetchLocations();
-    });
+    const searchString = event.target.value;
+    if(searchString){ 
+      this.onSuggestionsFetchRequested({searchString});
+    } else {
+      this.onSuggestionsClearRequested();
+    }
+  };
+
+  onBoundsChanged = ({center, radius}) => {
+    if(center.lat() === this.state.center.lat && 
+        center.lng() === this.state.center.lng) return;
+    this.fetchLocations(center, radius);
+  }
+
+  onSuggestionsFetchRequested = ({ searchString, reason }) => {
+    getOrganizations(searchString)
+      .then(organizations => { 
+          this.setState({ suggestions : organizations })
```

Comment:
Would be good to have a TODO here too about how this would be better in Redux state (like you have for locations).